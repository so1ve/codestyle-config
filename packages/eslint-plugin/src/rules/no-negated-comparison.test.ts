import { run } from "./_test";
import rule, { RULE_NAME } from "./no-negated-comparison";

const valid = ["a != b", "a !== b"];
const invalid = [
	"!(a == b)",
	"!(a === b)",
	"!(a != b)",
	"!(a !== b)",
	"!(a < b)",
	"!(a <= b)",
	"!(a > b)",
	"!(a >= b)",
];

run({
	name: RULE_NAME,
	rule,
	valid,
	invalid,
});
