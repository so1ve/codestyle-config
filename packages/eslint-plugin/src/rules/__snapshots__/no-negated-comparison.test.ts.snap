// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`no-negated-comparison > invalid > Invalid #0: !(a == b) 1`] = `"(a!=b)"`;

exports[`no-negated-comparison > invalid > Invalid #1: !(a === b) 1`] = `"(a!==b)"`;

exports[`no-negated-comparison > invalid > Invalid #2: !(a != b) 1`] = `"(a==b)"`;

exports[`no-negated-comparison > invalid > Invalid #3: !(a !== b) 1`] = `"(a===b)"`;

exports[`no-negated-comparison > invalid > Invalid #4: !(a < b) 1`] = `"(a>=b)"`;

exports[`no-negated-comparison > invalid > Invalid #5: !(a <= b) 1`] = `"(a>b)"`;

exports[`no-negated-comparison > invalid > Invalid #6: !(a > b) 1`] = `"(a<=b)"`;

exports[`no-negated-comparison > invalid > Invalid #7: !(a >= b) 1`] = `"(a<b)"`;
